sotodlib/core/metadata/loader.py

```diff
@@ -452,22 +491,42 @@ def check_tags(det_info, aug_request, final=False):
 
             # Make everything an axisman.
             if isinstance(item, ResultSet):
+                # Note this might raise an IncompleteDetInfoError.
                 item = broadcast_resultset(item, det_info=det_info)
 
             elif not isinstance(item, core.AxisManager):
                 logger.error(
                     f'The decoded item {item} is not an AxisManager or '
                     f'other well-understood type.  Request was: {request}.')
 
+            # You have to check for detector loss here -- compare
+            # item.dets.vals to what's in det_info.
+            i0 = core.util.get_multi_index(
+                item.dets.vals, det_info['readout_id'])
+
+            n_dets_item = len(set(i0[i0>=0]))
+            if n_dets_item < n_dets:
```

Comment:
Could we have this still fail if `on_missing == 'trim'` but `n_dets_item==0`? It doesn't really make sense to me to allow calculations to continue when we've cut all detectors and that also seems to be the most common failure mode folks are noticing. An error message that says which metadata is cutting all down to zero will be quite helpful.

Reply:
I don't know about this ... in that case it will already print a warning message that includes the metadata spec and the change in detector count.  The "computations to continue" should check, for themselves, that the data aren't useless for their purposes.

I don't like making little exceptions that hamstring general rules.  If we added this, then there would be *no way* to plow through with trim, if someone did want to do that.  Trim would sometimes trim, and sometimes fail... and that's an additional check (rather than one fewer check) that needs to be done in the calling code.

Reply:
ya ok. I do think this will be the largest user-reported "error that's not an error" but suppose that just requires training for folks.